evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -85,6 +85,9 @@ pub struct Cheatcodes {
 
     /// Scripting based transactions
     pub broadcastable_transactions: VecDeque<TypedTransaction>,
+
+    //// Offsets for files opened for reading
+    pub file_reading_offsets: HashMap<String, usize>,
```

Comment:
would it make sense to keep the actual File buf here?

but I guess that's currently not possible due to how `Cheatcodes` inspector is created (Clone)

Reply:
that's how I initially did it but yeah, due to `Clone` derive it's not possible unless we implement it on our own with some hashmap of bufs handling

Reply:
I faced similar issues in #2035

perhaps a better way would be to make two types

rn we have the `Inspectorstack` which is sort of like factory that can create new `Cheatcodes` inspectors, merely by cloning its `cheatcodes` field, because it can be initialized I think?

what if we replace clone with add dedicated `create_cheatcodes()` that clones the initialised subset and creates a new `Cheatcodes` object with an empty `Context` or something

I hope that made sense

```rust
impl Inspectorstack
{
  fn create_cheatcodes(&self) -> Cheatcodes {
    let subset = self.cheatcodes.clone();
    // merge all fields of subset to `Cheatcodes`
    Cheatcodes {... , context: Default::default()}
  }
}

```

Reply:
Yeah the reason we have `InspectorStack` cloneable is because we can configure the behavior of the inspectors (e.g. what inspectors are enabled/disabled). It also keeps an instance of `Cheatcodes` because some state needs to be persisted across some test boundaries, e.g. if you call `startPrank` in `setUp` this needs to apply for every test in that test contract.

Not sure if there's a better way to do that but I'd be keen to hear if there is because I don't really like how it works rn either.

Reply:
> what if we replace clone with add dedicated create_cheatcodes() that clones the initialised subset and creates a new Cheatcodes object with an empty Context or something

yeah, that sounds good. Do I get it right that no other fields except `file_reading_offsets`/`opened_read_files` should be in that `Context` struct? I thought `corrected_nonce` might be a candidate but its resetting looks better in broadcast-related piece:
https://github.com/foundry-rs/foundry/blob/54f102bb601b5712fc7f480d5bf25ae851b745f6/evm/src/executor/mod.rs#L370-L384

Reply:
also, it changes behaviour of file reading a bit: previously I guess it was possible to start reading file in one test, and continue in another (given they're executed in order). Now, each test has its own set of files opened for reading and it's not shared with other tests.